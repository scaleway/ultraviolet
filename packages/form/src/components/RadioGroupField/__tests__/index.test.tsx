import { screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm } from '@utils/test'
import { describe, expect, test, vi } from 'vitest'
import { RadioGroupField } from '..'

describe('RadioField', () => {
  test('should render correctly checked', async () => {
    const { asFragment } = renderWithForm(
      <RadioGroupField
        value="value-1"
        onChange={() => {}}
        name="radio"
        legend="Label"
      >
        <RadioGroupField.Radio name="value-1" value="value-1" label="Radio 1" />
        <RadioGroupField.Radio name="value-2" value="value-2" label="Radio 2" />
      </RadioGroupField>,
    )
    const [firstInput, secondInput] = screen.getAllByRole('radio', {
      hidden: true,
    })
    await userEvent.click(secondInput)

    expect(firstInput).not.toBeChecked()
    expect(secondInput).toBeChecked()

    expect(asFragment()).toMatchSnapshot()
  })

  test('should trigger events correctly', async () => {
    const onChange = vi.fn(() => {})

    const { asFragment } = renderWithForm(
      <RadioGroupField
        name="test"
        value="value-2"
        onChange={onChange}
        legend="RadioGroupField events"
      >
        <RadioGroupField.Radio name="value-1" value="value-1" label="Radio 1" />
        <RadioGroupField.Radio name="value-2" value="value-2" label="Radio 2" />
      </RadioGroupField>,
    )
    const input = screen.getAllByRole('radio', { hidden: true })[0]
    await userEvent.click(input)
    expect(onChange).toBeCalledTimes(1)
    expect(asFragment()).toMatchSnapshot()
  })
})
