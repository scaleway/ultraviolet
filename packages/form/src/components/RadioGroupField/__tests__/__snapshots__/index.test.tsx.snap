// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`radioField > should render correctly checked 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: contents;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-4 {
  border: none;
  padding: 0;
  margin: 0;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.75rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.emotion-11 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: text;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.emotion-17 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 0.5rem;
}

.emotion-17[aria-disabled='false'],
.emotion-17[aria-disabled='false']>label {
  cursor: pointer;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20+.emotion-22 {
  fill: #8c40ef;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20+.emotion-22 .emotion-24 {
  fill: #e5dbfd;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20[aria-invalid='true']+.emotion-22 {
  fill: #b3144d;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20[aria-invalid='true']+.emotion-22 .emotion-24 {
  fill: #ffd3e3;
}

.emotion-17[aria-disabled='true'] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-17[aria-disabled='true']>label,
.emotion-17[aria-disabled='true'] .emotion-20 {
  cursor: not-allowed;
}

.emotion-17[aria-disabled='true'] .emotion-22 {
  fill: #e9eaeb;
  cursor: not-allowed;
}

.emotion-17[aria-disabled='true'] .emotion-22 .emotion-24 {
  fill: #f3f3f4;
}

.emotion-19 {
  cursor: pointer;
  position: absolute;
  height: 1.5rem;
  width: 1.5rem;
  opacity: 0;
  white-space: nowrap;
  border-width: 0;
}

.emotion-19+.emotion-22 .emotion-26 {
  transform-origin: center;
  -webkit-transition: 200ms -webkit-transform ease-in-out;
  transition: 200ms transform ease-in-out;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.emotion-19:checked+svg .emotion-26 {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-19:checked[aria-disabled='false'][aria-invalid='false']+.emotion-22 {
  fill: #8c40ef;
}

.emotion-19:checked[aria-disabled='true'][aria-invalid='false']+.emotion-22 {
  fill: #d8c5fc;
}

.emotion-19[aria-invalid='true']:not([aria-disabled='true'])+.emotion-22 {
  fill: #e51963;
}

.emotion-19[aria-disabled='false']:active+.emotion-22 {
  background-color: #5e127e40;
  fill: #8c40ef;
}

.emotion-19[aria-disabled='false']:active+.emotion-22 .emotion-24 {
  fill: #f1eefc;
}

.emotion-19[aria-disabled='false']:focus-visible+.emotion-22 {
  outline: -webkit-focus-ring-color auto 1px;
}

.emotion-19[aria-invalid='true']:focus+.emotion-22 {
  background-color: #f91b6c40;
  fill: #e51963;
}

.emotion-19[aria-invalid='true']:focus+.emotion-22 .emotion-24 {
  fill: #ffebf2;
}

.emotion-21 {
  height: 1.5rem;
  width: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  border-radius: 100%;
  fill: #d9dadd;
}

.emotion-21 .emotion-24 {
  fill: #ffffff;
}

.emotion-28 {
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

<div
    data-testid="testing"
  >
    <form
      class="emotion-0 emotion-1"
      novalidate=""
    >
      <div
        class="emotion-2 emotion-3"
      >
        <fieldset
          class="emotion-4 emotion-5"
        >
          <div
            class="emotion-6 emotion-3"
          >
            <div
              class="emotion-8 emotion-3"
            >
              <legend
                class="emotion-10 emotion-11 emotion-12"
              >
                Label
              </legend>
            </div>
            <div
              class="emotion-13 emotion-3"
            >
              <div
                class="emotion-15 emotion-16"
              >
                <div
                  aria-disabled="false"
                  class="emotion-17 emotion-18"
                  data-checked="false"
                  data-error="false"
                >
                  <input
                    aria-disabled="false"
                    aria-invalid="false"
                    class="emotion-19 emotion-20"
                    id="_r_1_"
                    name="radio"
                    type="radio"
                    value="value-1"
                  />
                  <svg
                    class="emotion-21 emotion-22"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                        stroke-width="2"
                      />
                      <circle
                        class="emotion-23 emotion-24"
                        cx="12"
                        cy="12"
                        r="8"
                      />
                      <circle
                        class="emotion-25 emotion-26"
                        cx="12"
                        cy="12"
                        r="5"
                      />
                    </g>
                  </svg>
                  <label
                    class="emotion-27 emotion-28 emotion-12"
                    for="_r_1_"
                  >
                    Radio 1
                  </label>
                </div>
              </div>
              <div
                class="emotion-15 emotion-16"
              >
                <div
                  aria-disabled="false"
                  class="emotion-17 emotion-18"
                  data-checked="true"
                  data-error="false"
                >
                  <input
                    aria-disabled="false"
                    aria-invalid="false"
                    class="emotion-19 emotion-20"
                    id="_r_4_"
                    name="radio"
                    type="radio"
                    value="value-2"
                  />
                  <svg
                    class="emotion-21 emotion-22"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                        stroke-width="2"
                      />
                      <circle
                        class="emotion-23 emotion-24"
                        cx="12"
                        cy="12"
                        r="8"
                      />
                      <circle
                        class="emotion-25 emotion-26"
                        cx="12"
                        cy="12"
                        r="5"
                      />
                    </g>
                  </svg>
                  <label
                    class="emotion-27 emotion-28 emotion-12"
                    for="_r_4_"
                  >
                    Radio 2
                  </label>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`radioField > should trigger events correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: contents;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-4 {
  border: none;
  padding: 0;
  margin: 0;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.75rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.emotion-11 {
  color: #3f4250;
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 500;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: text;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.5rem;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: normal;
  -webkit-box-align: normal;
  -ms-flex-align: normal;
  align-items: normal;
  -webkit-box-pack: normal;
  -ms-flex-pack: normal;
  -webkit-justify-content: normal;
  justify-content: normal;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.25rem;
}

.emotion-17 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  gap: 0.5rem;
}

.emotion-17[aria-disabled='false'],
.emotion-17[aria-disabled='false']>label {
  cursor: pointer;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20+.emotion-22 {
  fill: #8c40ef;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20+.emotion-22 .emotion-24 {
  fill: #e5dbfd;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20[aria-invalid='true']+.emotion-22 {
  fill: #b3144d;
}

.emotion-17:hover[aria-disabled='false'] .emotion-20[aria-invalid='true']+.emotion-22 .emotion-24 {
  fill: #ffd3e3;
}

.emotion-17[aria-disabled='true'] {
  cursor: not-allowed;
  color: #b5b7bd;
}

.emotion-17[aria-disabled='true']>label,
.emotion-17[aria-disabled='true'] .emotion-20 {
  cursor: not-allowed;
}

.emotion-17[aria-disabled='true'] .emotion-22 {
  fill: #e9eaeb;
  cursor: not-allowed;
}

.emotion-17[aria-disabled='true'] .emotion-22 .emotion-24 {
  fill: #f3f3f4;
}

.emotion-19 {
  cursor: pointer;
  position: absolute;
  height: 1.5rem;
  width: 1.5rem;
  opacity: 0;
  white-space: nowrap;
  border-width: 0;
}

.emotion-19+.emotion-22 .emotion-26 {
  transform-origin: center;
  -webkit-transition: 200ms -webkit-transform ease-in-out;
  transition: 200ms transform ease-in-out;
  -webkit-transform: scale(0);
  -moz-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
}

.emotion-19:checked+svg .emotion-26 {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-19:checked[aria-disabled='false'][aria-invalid='false']+.emotion-22 {
  fill: #8c40ef;
}

.emotion-19:checked[aria-disabled='true'][aria-invalid='false']+.emotion-22 {
  fill: #d8c5fc;
}

.emotion-19[aria-invalid='true']:not([aria-disabled='true'])+.emotion-22 {
  fill: #e51963;
}

.emotion-19[aria-disabled='false']:active+.emotion-22 {
  background-color: #5e127e40;
  fill: #8c40ef;
}

.emotion-19[aria-disabled='false']:active+.emotion-22 .emotion-24 {
  fill: #f1eefc;
}

.emotion-19[aria-disabled='false']:focus-visible+.emotion-22 {
  outline: -webkit-focus-ring-color auto 1px;
}

.emotion-19[aria-invalid='true']:focus+.emotion-22 {
  background-color: #f91b6c40;
  fill: #e51963;
}

.emotion-19[aria-invalid='true']:focus+.emotion-22 .emotion-24 {
  fill: #ffebf2;
}

.emotion-21 {
  height: 1.5rem;
  width: 1.5rem;
  min-width: 1.5rem;
  min-height: 1.5rem;
  border-radius: 100%;
  fill: #d9dadd;
}

.emotion-21 .emotion-24 {
  fill: #ffffff;
}

.emotion-28 {
  font-size: 1rem;
  font-family: Inter,sans-serif;
  font-weight: 400;
  letter-spacing: 0;
  line-height: 1.5rem;
  text-transform: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  cursor: pointer;
}

<div
    data-testid="testing"
  >
    <form
      class="emotion-0 emotion-1"
      novalidate=""
    >
      <div
        class="emotion-2 emotion-3"
      >
        <fieldset
          class="emotion-4 emotion-5"
        >
          <div
            class="emotion-6 emotion-3"
          >
            <div
              class="emotion-8 emotion-3"
            >
              <legend
                class="emotion-10 emotion-11 emotion-12"
              >
                RadioGroupField events
              </legend>
            </div>
            <div
              class="emotion-13 emotion-3"
            >
              <div
                class="emotion-15 emotion-16"
              >
                <div
                  aria-disabled="false"
                  class="emotion-17 emotion-18"
                  data-checked="true"
                  data-error="false"
                >
                  <input
                    aria-disabled="false"
                    aria-invalid="false"
                    class="emotion-19 emotion-20"
                    id="_r_8_"
                    name="test"
                    type="radio"
                    value="value-1"
                  />
                  <svg
                    class="emotion-21 emotion-22"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                        stroke-width="2"
                      />
                      <circle
                        class="emotion-23 emotion-24"
                        cx="12"
                        cy="12"
                        r="8"
                      />
                      <circle
                        class="emotion-25 emotion-26"
                        cx="12"
                        cy="12"
                        r="5"
                      />
                    </g>
                  </svg>
                  <label
                    class="emotion-27 emotion-28 emotion-12"
                    for="_r_8_"
                  >
                    Radio 1
                  </label>
                </div>
              </div>
              <div
                class="emotion-15 emotion-16"
              >
                <div
                  aria-disabled="false"
                  class="emotion-17 emotion-18"
                  data-checked="false"
                  data-error="false"
                >
                  <input
                    aria-disabled="false"
                    aria-invalid="false"
                    class="emotion-19 emotion-20"
                    id="_r_b_"
                    name="test"
                    type="radio"
                    value="value-2"
                  />
                  <svg
                    class="emotion-21 emotion-22"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <circle
                        cx="12"
                        cy="12"
                        r="10"
                        stroke-width="2"
                      />
                      <circle
                        class="emotion-23 emotion-24"
                        cx="12"
                        cy="12"
                        r="8"
                      />
                      <circle
                        class="emotion-25 emotion-26"
                        cx="12"
                        cy="12"
                        r="5"
                      />
                    </g>
                  </svg>
                  <label
                    class="emotion-27 emotion-28 emotion-12"
                    for="_r_b_"
                  >
                    Radio 2
                  </label>
                </div>
              </div>
            </div>
          </div>
        </fieldset>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
